setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php

```diff
@@ -16,11 +16,25 @@ class ClassesScanner implements ClassesScannerInterface
     protected $excludePatterns = [];
 
     /**
+     * @var array
+     */
+
+    protected $fileResults = [];
+
+    /**
+     * @var DirectoryList
+     */
+
+    protected $directoryList;
+
+    /**
+     * @param DirectoryList $directoryList
      * @param array $excludePatterns
      */
-    public function __construct(array $excludePatterns = [])
+    public function __construct(DirectoryList $directoryList, array $excludePatterns = [])
```

Comment:
To increase the chance of getting this released as soon as possible, please make new constructor arguments optional so that there is no backwards incompatibility. Then if it is null, use `ObjectManager::getInstance()->get()` to retrieve the object (yes, that's the recommended  practice)